cdap-unit-test/src/test/java/co/cask/cdap/test/app/AppWithServices.java

```diff
@@ -243,6 +246,13 @@ public void stop() {
           public void run(DatasetContext context) throws Exception {
             KeyValueTable table = context.getDataset(DATASET_NAME);
             table.write(DATASET_TEST_KEY_STOP, valueToWriteOnStop);
+
+            // Get another instance of the dataset with different runtime arguments and write a new value
+            // only if its a different instance
+            KeyValueTable table2 = context.getDataset(DATASET_NAME, ImmutableMap.of("arg", "value"));
```

Comment:
It would be a good idea to get the Dataset again with the same arguments to make sure you get the same Dataset.